File: practise/leetcode/DailyQuestion/202406/575.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        map<int,int> ctmp;
        for(auto e:candyType) ctmp[e]++;
        int ans=candyType.size()/2;
        ans=min(ans,(int)ctmp.size());
        return ans;
    }
};

class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        unordered_set<int> s(candyType.begin(), candyType.end());
        return min(s.size(), candyType.size() / 2);
    }
};

File: practise/leetcode/DailyQuestion/202406/575.py
class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        return min(len(set(candyType)), len(candyType) // 2)

class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        ctmp = {}
        for e in candyType:
            if e in ctmp:
                ctmp[e] += 1
            else:
                ctmp[e] = 1
        ans = len(candyType) // 2
        return min(ans, len(ctmp))